b/c/d)
# Z-Test of Two Means Independent Groups

The FIFA Football (Soccer) World Cup is held every four years and is one of the biggest sporting events in the world. In 2006, Germany hosted the World Cup. A study was conducted by Dr. Wilbert-Lampen, et. al. to determine if the stress of viewing a soccer match would increase the risk of a heart attack or another cardiovascular event.
We will use the data on cardiovascular problems during the World Cup to test the hypothesis that the mean number of cardiovascular events is greater during the World Cup than during the control period. (from https://byuistats.github.io/BYUI_M221_Book/Lesson13.html)



In [46]:
import pandas as pd

#Import data from Excel sheet
df=pd.read_excel(r'C:\DS350_W24\DS350_WI24_Blad_Alex\week_01\WorldCupHeartAttacks.xlsx')

print(df)

#Get the Means
x1=df['HeartAttacks1'].mean()
x2=df['HeartAttacks2'].mean()

print(x1,x2)

#Std. Deviations
s1=df['HeartAttacks1'].std()
s2=df['HeartAttacks2'].std()

print(s1,s2)

#Sample Size
n1=df['HeartAttacks1'].count()
n2=df['HeartAttacks2'].count()

print(n1, n2)

dif=x1-x2

                   Date1  HeartAttacks1 TimePeriod1        Date2  \
0    2003-05-01 00:00:00             12     Control  May 1, 2006   
1            May 2, 2003             12     Control  May 2, 2006   
2            May 3, 2003              6     Control  May 3, 2006   
3            May 4, 2003             16     Control  May 4, 2006   
4            May 5, 2003             13     Control  May 5, 2006   
..                   ...            ...         ...          ...   
177        July 26, 2005             15     Control          NaN   
178        July 27, 2005             16     Control          NaN   
179        July 28, 2005             22     Control          NaN   
180        July 29, 2005              7     Control          NaN   
181        July 30, 2005             12     Control          NaN   

     HeartAttacks2 TimePeriod2  
0             16.0   World Cup  
1             19.0   World Cup  
2             13.0   World Cup  
3             20.0   World Cup  
4             26.0

Now we can calculate the z-score by hand.


In [43]:
import numpy as np

z= (x1-x2)/np.sqrt((s1**2/n1)+(s2**2/n2))
print(z)

-4.617222713734024


The test statistic found by hand we can approximate to be -4.617, now we'll try to get the same statistic using the scipy.

In [68]:
from scipy import stats

d1=df['HeartAttacks1'].to_numpy()

d2=df['HeartAttacks2'].to_numpy()


print(d1)
print(d2)

stats.ttest_ind(d1,d2, 0 ,False, 'omit')


[12 12  6 16 13 17 14 16 12 12 16 13 13 14 18 11 11 14  6 13 23 15 17 17
 18 13 20 10  8 21 10  6 11  8 12 11 18 15 17 17 10 16 13 17 11  5 18 15
 23 16  9 13 18 12 16 10 11 15  8 21 10 13 18  9 17 10 17 11 16  9 16 23
 13 13  9 10  6 14 19 13 16 18 17 18 16 14 17 19 27 11 18 23 12 19 11 11
 15 10 19 17 16 15 10  9  7 12 12 20 16 20  9  8 22 14 17 19 15 10 14 17
 20 14 14 15 11  8 13 13 27 17 15 10 14 14 10 14  9 17  9  8 15 11 21 14
 14 13  4 10 15 22 12 17 15 16 12 11 17 15 13 14 14 10 15 18 13 16 16  9
 19 12 11 15 16 11 13 15  9 15 16 22  7 12]
[16. 19. 13. 20. 26.  5. 23. 17. 11. 12. 25. 22. 19. 16. 13. 15. 16. 13.
 16. 14. 15. 23. 16. 17. 15. 10. 13. 11. 19. 17. 10. 14. 12. 10. 10. 22.
 33. 26. 20. 43. 13. 21. 22. 20. 49. 11. 26. 18. 15. 23. 31. 21. 11. 15.
 39. 18. 18. 15. 19. 15. 63. 23. 19. 15. 62. 22. 14. 12. 15. 32.  9. 23.
 11. 13. 21. 15. 20. 20. 26. 14. 21. 28. 13. 12. 12. 18. 19. 20.  8.  7.
 15. nan nan nan nan nan nan nan nan nan nan nan nan nan nan nan nan nan
 nan na

Ttest_indResult(statistic=-4.617222713734024, pvalue=1.0927209957246354e-05)

This gives us the same value for the test statistic calculated by hand earlier!

# Z-Test of Single Proportion 

Suppose we are interested in estimating the proportion of individuals in a population who have a certain trait. For instance, we might be interested in studying the proportion of children living near a lead smelter who have colic. The prevalence of colic in the general public is estimated to be as low as 7%.


From https://sphweb.bumc.bu.edu/otlt/mph-modules/bs/sas/sas6-categoricaldata/SAS6-CategoricalData2.html#headingtaglink_3 
a sample was take with n=124 and the number of kids with colic x=23. With this information we see that 

n=124
x=23 
p0=.07 

and the point estimate p, give by p=x/n is
p=.19

Doing the Z-test of a single proportion "by hand we see"

In [26]:
import numpy as np 

n=124
x=23
p0=.07
p=.19
q0=.93

z_score= (p-p0)/np.sqrt((p0*q0)/n)
print(z_score)


5.237229365663816


We can also use the proportions_ztest() function from the statsmodels library 
and it gives us the following. 

In [25]:
from statsmodels.stats.proportion import proportions_ztest

proportions_ztest(count=23, nobs=124, value=.07, alternative='larger')

(3.3084856526541473, 0.00046900995453509457)

Although the z-value is differnt, it still leads us to reject the null-hypothesis. I assume that the function may calculate the z-statistic differently than the give formula that we have. 

e) I have found this data set about a sample of apples. It could be used to Test the catagorical frequency of the quality of the apple, or to do a z-test of the mean of the weight of an apple within the sample population. 

https://www.kaggle.com/datasets/nelgiriyewithana/apple-quality